pkg/plugin/datasource.go

```diff
@@ -213,7 +291,25 @@ func (ds *TwinMakerDatasource) DoQuery(ctx context.Context, query models.TwinMak
 	case models.QueryTypeGetPropertyValue:
 		return ds.handler.GetPropertyValue(ctx, query)
 	case models.QueryTypeEntityHistory:
-		return ds.handler.GetEntityHistory(ctx, query)
+		res := ds.handler.GetEntityHistory(ctx, query)
+		frame := data.NewFrame("response")
+
+		// Add fields (matching the same schema used in QueryData).
+		frame.Fields = append(frame.Fields,
+			data.NewField("values", nil, make([]float64, 1)),
+			data.NewField("time", nil, make([]time.Time, 1)),
+		)
+		res.Frames = append(res.Frames, frame)
+		res.Error = nil
+		for i,_ := range res.Frames {
+			res.Frames[i].SetMeta(&data.FrameMeta{
+				Channel: fmt.Sprintf("ds/%s/30s/%s/%s", ds.settings.UID, query.EntityId, query.ComponentName),
```

Comment:
is this still needed with the new changes to `datasource.ts`?

Reply:
ah no I'll clean up